### ISSUE TITLE
Error in type annotation of NumberFormatBaseProps (NumberFormatBase)

### ISSUE THREAD

**Issue by akhmadullin** [s-yadav/react-number-format#740]
#### Describe the issue and the actual behavior
`NumberFormatBase` component's props have following [type annotation](https://github.com/s-yadav/react-number-format/blob/master/src/types.ts#L60):

![Снимок экрана 2023-03-06 в 12 35 21](https://user-images.githubusercontent.com/37641303/223006743-f99fc48c-3ce9-4506-b95b-236402e47287.png)

As you can see, `format`, `removeFormatting`  and `getCaretBoundary` props are defined as required props. 

But in [implementation of `NumberFormatBase` component](https://github.com/s-yadav/react-number-format/blob/master/src/number_format_base.tsx#L30) we can see, that these props are optional, because they have default value:

![Снимок экрана 2023-03-06 в 12 47 19](https://user-images.githubusercontent.com/37641303/223008145-56492513-6995-47ad-b818-9bf635944495.png)

Because `format`, `removeFormatting`  and `getCaretBoundary` props are defined as required, I have following TS error*, when I try to use `NumberFormatBase` component with only one set prop - `format`:

![Снимок экрана 2023-03-06 в 12 57 18](https://user-images.githubusercontent.com/37641303/223009936-57dcc426-5622-4072-97a3-cb3d6e058d7a.png)

*TS error - `Type '{ valueIsNumericString: true; value: string; onValueChange: (values: NumberFormatValues) => void; format: (phoneNumber: string) => string; }' is missing the following properties from type 'NumberFormatBase': removeFormatting, getCaretBoundary ts(2739)`

I set only `format` prop, because I need only custom formatting in my case. But TS expects `removeFormatting`  and `getCaretBoundary` will be set too. Despite the fact, that  `removeFormatting`  and `getCaretBoundary` props are missing, component works ok, because `NumberFormatBase` component has default values for these props.

#### Describe the expected behavior
I see 2 possible solutions:

1) `format`, `removeFormatting`  and `getCaretBoundary` props are defined as optional props in type annotation:

```typescript
type NumberFormatBase = {
  format?: FormatInputValueFunction;
  removeFormatting?: RemoveFormattingFunction;
  getCaretBoundary?: (formattedValue: string) => boolean[];
};
```

2) If `react-number-format` for some reason needs  `format`, `removeFormatting`  and `getCaretBoundary` props to be required,  default values of these props should be exported too, to be used where lib users need default behavior:

```typescript
// react-number-format
export { caretUnknownFormatBoundary, defaultRemoveFormatting, defaultFormat }
```

```typescript
// some project
import { NumberFormatBase, caretUnknownFormatBoundary, defaultRemoveFormatting  } from 'react-number-format';

const customFormatter = () => {...};

...

return (
    <NumberFormatBase 
        format={customFormatter}
        removeFormatting={defaultRemoveFormatting}
        getCaretBoundary={caretUnknownFormatBoundary}
    />
)
```

#### Provide a CodeSandbox link illustrating the issue

Modified demo from your docs - https://codesandbox.io/s/card-expiry-field-forked-8ex966?file=/src/App.tsx

**Comment by danielweil** [comment]
UPVOTE

**Comment by s-yadav** [comment]
Yes, make sense. Will fix the annotation here.

**Comment by s-yadav** [comment]
This is fixed on v5.2.0

**Comment by frindly** [comment]
This produces a breaking change in typescript for the  useNumericFormat hook 

Consider: 


```
const { format, removeFormatting } = useNumericFormat(NumericFormatCurrencyProps);
...
format(val);

```

This update causes these newly optional properties to be returned as FormatInputValueFunction | undefined, so typescript will now complain with `TS2722: Cannot invoke an object which is possibly 'undefined'.`, Is this hook returning undefined fns for these values actually possible?
